core/Network/TLS/Struct.hs

```diff
@@ -566,3 +586,18 @@ instance EnumSafe16 Group where
     toEnumSafe16 259 = Just FFDHE6144
     toEnumSafe16 260 = Just FFDHE8192
     toEnumSafe16 _   = Nothing
+
+
+-- Digital signature algorithm, in close relation to public/private key types
+-- and cipher key exchange.
+data DigitalSignatureAlg = RSA | DSS | ECDSA | Ed25519 | Ed448
+                           deriving (Show, Eq)
+
```

Comment:
I don't like DigitalSignatureAlg being defined in module "Struct" as it's internal and not protocol-related. Module "Types" would be better perhaps.

And if possible signatureCompatible should remain in "Handshake.Signature" where signatureParams is also defined (because those two must stay in sync).